src/oneD/Boundary1D.cpp

```diff
@@ -610,6 +610,30 @@ ReactingSurf1D::ReactingSurf1D()
     m_type = cSurfType;
 }
 
+ReactingSurf1D::ReactingSurf1D(shared_ptr<Solution> solution)
+{
+    if (!std::dynamic_pointer_cast<SurfPhase>(solution->thermo())) {
+        throw CanteraError("ReactingSurf1D::ReactingSurf1D",
+            "Detected incompatible ThermoPhase type '{}'", solution->thermo()->type());
+    }
+    if (!std::dynamic_pointer_cast<InterfaceKinetics>(solution->kinetics())) {
+        throw CanteraError("ReactingSurf1D::ReactingSurf1D",
+            "Detected incompatible kinetics type '{}'",
+            solution->kinetics()->kineticsType());
+    }
+    m_solution = solution;
+    m_kin = (InterfaceKinetics*)solution->kinetics().get();
+    m_surfindex = m_kin->surfacePhaseIndex();
+    m_sphase = (SurfPhase*)&m_kin->thermo(m_surfindex);
```

Comment:
To get rid of the C-style casts (which I would like to avoid), you could assign `m_kin` and `m_sphase` using `dynamic_pointer_cast` and immediately after each do the check that they are non-zero.

Reply:
Done. I also further simplified the logic, as some of these checks were redundant.